[#55961] Define the project lists query for activated storages

judithroth opened the pull request:
https://community.openproject.org/work_packages/55961

This also defines a basic table in the view, so it can be tested. I've choosen not to go with a new query filter for projects belonging to a storage because it seemed to be more work than necessary and not provide any further benefits, since for the API there already is such a filter (which also seems to be tailored to the needs of the API. It would not have worked for the table because the table should include archived projects as well).

judithroth committed:
[#55961] Define the project lists query for activated storages

https://community.openproject.org/work_packages/55961

This also defines a basic table in the view, so it can be tested.
I've choosen not to go with a new query filter for projects belonging
to a storage because it seemed to be more work than necessary and not
provide any further benefits, since for the API there already is
such a filter (which also seems to be tailored to the needs of the
API. It would not have worked for the table because the table should
include archived projects as well).

judithroth commented on modules/storages/app/views/storages/admin/storages/project_mappings.html.erb:
```diff
@@ -34,3 +34,10 @@
+<%=
+  render(Storages::ProjectStorages::ProjectMapping::TableComponent.new(
+    query: @project_query,
+    params: params
```
@akabiru I assume this is a false positive and could be added to brakeman.ignore, but I am not sure about it. Do you have a suggestion on how to handle this?

akabiru reviewed (changes requested):
Looking good Judith! ⭐ 

🟠 Could we add https://github.com/opf/openproject/pull/16018 first into `dev` first and rebase this PR so it fits into the new structure?

akabiru commented on modules/storages/app/components/storages/project_storages/project_mapping/row_component.rb:
```diff
@@ -0,0 +1,34 @@
+
+# Purpose: Defines how to format the components within a table row of Projects
+# associated with a Storage
+module Storages::ProjectStorages::ProjectMapping
```
🟢 Given the new namespacing structure in https://github.com/opf/openproject/pull/16018 I think the following would be more fitting- as we essentially can drop the "project_mapping" naming across the board
```suggestion
module Storages::ProjectStorages
```

akabiru replied:
i.e. we'd have

1. `Storages::ProjectStorages::ProjectRowComponent` - for  projects
2. `Storages::ProjectStorages::StorageRowComponent` - for  storages _(in the future when the [current list is primerised](https://qa.openproject-edge.com/projects/demo-project/settings/project_storages/external_file_storages))_

judithroth replied:
I'd rather go with another namespace here. It's also nice if the files are grouped (not only by name but by a folder). What do you think about
`Storages::ProjectStorages::Projects::RowComponent` (and `Storages::ProjectStorages::Projects::TableComponent`) and keeping the other files as is?

akabiru replied:
That also works- +1 on folder naming, can be further reused 👍🏾

akabiru commented on modules/storages/app/components/storages/project_storages/project_mapping/table_component.rb:
```diff
@@ -0,0 +1,48 @@
+# Projects for a given Storage.
+# See also: row_component.rb, which contains a method
+# for every "column" defined below.
+module Storages::ProjectStorages::ProjectMapping
```
🟢 ditto:
```suggestion
module Storages::ProjectStorages
```

judithroth commented:
> Looking good Judith! ⭐
> 
> 🟠 Could we add #16018 first into `dev` first and rebase this PR so it fits into the new structure?

Sure! - I thought the same actually so I can adjust wording and some file names :slightly_smiling_face:

judithroth committed:
[#55961] Adapt to new controller and naming

Which was introduced with [#55966] Define "Enabled in Projects" as a fully qualified nested resource / pull request #16018

akabiru commented on modules/storages/app/components/storages/project_storages/projects/table_component.rb:
```diff
@@ -0,0 +1,48 @@
+  class TableComponent < Projects::TableComponent # rubocop:disable OpenProject/AddPreviewForViewComponent
+    include OpTurbo::Streamable
+
+    def sortable?
+      false
+    end
```
Oddly enough this doesn't seem to kick in:

- [ ] I still see the "sorting icon" - which should be disabled in this feature.
- [ ] The action list is filled in with the default properties, in this feature spec- the menu actions are (1) Edit Project Folder (2) Remove project - I suggest either leaving them empty or defining placeholders.

![Screenshot 2024-07-05 at 11 15 57 AM](https://github.com/opf/openproject/assets/17295175/9233c431-9daa-4b45-8b92-4f96df029e99)

akabiru replied:
https://github.com/opf/openproject/assets/17295175/b2a0d89b-4261-4aaf-b8b5-2925c13161f4

akabiru reviewed (commented):
🟠 Almost there - two issues mentioned in the comments. 🛠️

akabiru reviewed (approved):
:shipit:

akabiru commented:
We agreed to address the issues separately 👍🏾
